File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!= len(t):
            return False

        count = {}
        count1= {}
        for ch in s:
            count[ch] = count.get(ch,0) + 1
        
        for ch in t:
            count1[ch] = count1.get(ch,0) + 1
        if count == count1:
            return True
        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const map = new Map();

        for(const num of nums){
            map.set(num,(map.get(num)||0)+1);
        }

        const buckets = Array.from({ length: nums.length + 1},() => []);


        for (const [num,count] of map.entries()) {
            buckets[count].push(num);
        }

        const result = [];
        for(let i = buckets.length - 1 ; i >= 0; i--){
        for (const num of buckets[i]) {
            result.push(num);
            if( result.length === k) {
                return result;
            }
        }
        }

        return result
    }
}
